src/pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -143,10 +143,13 @@ def advance(self, *args: Any, **kwargs: Any) -> None:
         dataloader_idx = self.current_dataloader_idx
         dataloader = self.current_dataloader
         assert self._data_fetcher is not None
-        self._data_fetcher.setup(
-            dataloader,
-            batch_to_device=partial(self.trainer._call_strategy_hook, "batch_to_device", dataloader_idx=dataloader_idx),
-        )
+
+        def batch_to_device(batch: Any, dataloader_idx: int) -> Any:
+            batch = self.trainer.lightning_module._on_before_batch_transfer(batch, dataloader_idx=dataloader_idx)
+            batch = self.trainer._call_strategy_hook("batch_to_device", batch, dataloader_idx=dataloader_idx)
+            return batch
+
+        self._data_fetcher.setup(dataloader, batch_to_device=partial(batch_to_device, dataloader_idx=dataloader_idx))
```

Comment:
I don't think the `partial` is necessary anymore as it can be put inside the local function you just added above

```suggestion

        def batch_to_device(batch: Any) -> Any:
            batch = self.trainer.lightning_module._on_before_batch_transfer(batch, dataloader_idx=dataloader_idx)
            batch = self.trainer._call_strategy_hook("batch_to_device", batch, dataloader_idx=dataloader_idx)
            return batch

        self._data_fetcher.setup(dataloader, batch_to_device=batch_to_device
```